charts/external-dns/templates/_helpers.tpl

```diff
@@ -70,3 +70,30 @@ The image to use
 {{- define "external-dns.image" -}}
 {{- printf "%s:%s" .Values.image.repository (default (printf "v%s" .Chart.AppVersion) .Values.image.tag) }}
 {{- end }}
+
+
+{{/*
+Keep backward compatibility on provider
+*/}}
+{{- define "external-dns.providername" -}}
+{{- if eq (typeOf .Values.provider) "string" }}
+{{- .Values.provider }}
+{{- else }}
+{{- .Values.provider.name }}
+{{- end }}
+{{- end }}
+
+{{/*
+The webhook providers
+*/}}
+{{- define "external-dns.webhook-providers" -}}
+{{- if eq (typeOf .Values.provider) "object" }}
+{{- $version := default "latest" .Values.provider.version }}
+adguard: ghcr.io/muhlba91/external-dns-provider-adguard:{{ $version }}
```

Comment:
I would suggest avoiding hardcoded image names inside a _helpers.tpl as this is the last thing folks will look into and it would need to be synced regularly. Maybe we can create a map in the values.yaml to fetch this information from? (also allows users to override it easily)

Reply:
I concur. We should not be adding such mappings to the chart.

Reply:
Thanks for this suggestion. I moved them to values with 287eb931